pytensor/link/pytorch/dispatch/elemwise.py

```diff
@@ -34,3 +35,18 @@ def dimshuffle(x):
         return res
 
     return dimshuffle
+
+
+@pytorch_funcify.register(Softmax)
+def pytorch_funcify_Softmax(op, **kwargs):
+    axis = op.axis
+
+    if axis is None:
```

Comment:
Hmm, None means all axis in PyTensor, doesn't Pytorch support that case?

Reply:
None in Pytorch implicitly means axis 1: https://discuss.pytorch.org/t/implicit-dimension-choice-for-softmax-warning/12314/10

And that funcationality is deprecated now, giving the warning:

> UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  torch.nn.functional.softmax(torch.Tensor(x), dim=None)

Reply:
So can we write the function to achieve the same behavior as softmax along all axis? I guess in that case we need to do it manually with `exp(x) / torch.sum(exp(x), axis=None)` or is that also not allowed?

Reply:
Yes, reshaping can be used for `axis=None`
Also, it might be a good idea to make sure `x` is a `torch.float` dtype. The softmax would fail if the dtype is int or long.

Reply:
@ricardoV94 , what is the expected behaviour when the input is of dtype int?

I'm thinking of converting to float implicitly.

Reply:
Have to check the integer case but don't forget to test it as well

Reply:
Hi @ricardoV94, I have added the Logsoftmax and Softmaxgrad ops.

I have a question in relation to testing the types. How do I do this?

Everytime I send the functiongraph and the inputs into `compare_pytorch_and_py`, the function always ends up with a Tensor of floats somehow.

Reply:
In the test you're creating a `matrix` which has a default dtype of `float32` or `float64`, you can parametrize the test with an explicit dtype, say `pytest.mark.parametrize("dtype", ["float64", "int64"])` to test integer types as well.

This is something we should mention in the docs @HarshvirSandhu